secretsharing/ss.go

```diff
@@ -0,0 +1,183 @@
+// Package secretsharing provides methods to split secrets into shares.
+//
+// Let n be the number of parties, and t the number of corrupted parties such
+// that 0 <= t < n. A (t,n) secret sharing allows to split a secret into n
+// shares, such that the secret can be recovered from any subset of t+1 shares.
+//
+// The NewShamirSecretSharing function creates a Shamir secret sharing [1],
+// which relies on Lagrange polynomial interpolation.
+//
+// The NewFeldmanSecretSharing function creates a Feldman secret sharing [2],
+// which extends Shamir's by allowing to verify that a share is part of a
+// committed secret.
+//
+// In this implementation, secret sharing is defined over the scalar field of
+// a prime order group.
+//
+// References
+//
+//	[1] https://dl.acm.org/doi/10.1145/359168.359176
+//	[2] https://ieeexplore.ieee.org/document/4568297
+package secretsharing
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/cloudflare/circl/group"
+	"github.com/cloudflare/circl/math/polynomial"
+)
+
+// Share represents a share of a secret.
+type Share struct {
+	// ID uniquely identifies a share in a secret sharing instance.
+	ID group.Scalar
```

Comment:
Can we rename this to something else? Maybe `Input`, and maybe rename `Value` to `Output`?